heron/common/src/java/org/apache/heron/common/basics/ByteAmount.java

```diff
@@ -221,42 +202,37 @@ public ByteAmount max(ByteAmount other) {
     }
   }
 
-  @Override
-  public int compareTo(ByteAmount other) {
-    return Long.compare(asBytes(), other.asBytes());
-  }
-
   @Override
   public boolean equals(Object other) {
     if (this == other) {
       return true;
     }
-    if (other == null || getClass() != other.getClass()) {
+    if (!(other instanceof ByteAmount)) {
```

Comment:
null check is still needed.

Reply:
Looks like `instanceof` automatically does null check, so it's redundant

Reply:
The code below prints Yes.

     Thread x = null;
      if (x instanceof Thread) {
        System.out.println("Yes!");
      }

So when other is null, "if (!(other instanceof ByteAmount))" returns false, and .value will be called later.